From patchwork Tue Sep 11 14:58:38 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Paolo Bonzini <pbonzini@redhat.com>
X-Patchwork-Id: 10595771
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id E39EF14BD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 11 Sep 2018 14:58:44 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D2F92297A7
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 11 Sep 2018 14:58:44 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id C713029752; Tue, 11 Sep 2018 14:58:44 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.8 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,T_DKIM_INVALID autolearn=ham
 version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 768E7297BB
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 11 Sep 2018 14:58:44 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726800AbeIKT6W (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 11 Sep 2018 15:58:22 -0400
Received: from mail-wr1-f43.google.com ([209.85.221.43]:33843 "EHLO
        mail-wr1-f43.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726622AbeIKT6W (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 11 Sep 2018 15:58:22 -0400
Received: by mail-wr1-f43.google.com with SMTP id g33-v6so26438765wrd.1
        for <kvm@vger.kernel.org>; Tue, 11 Sep 2018 07:58:41 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=sender:from:to:subject:date:message-id;
        bh=CWCynePQ8um+5nRyCgMSCXhJ2HrwalLQWjg5pt3EE68=;
        b=icbVXG/z92Ky02899MJ2otziNjsbDZn5P1J2hhG4yBC/RZ3+OzOx9qCqYkGFvNJMmW
         JLLvOsw72PVwWvTJ2xCPa4Q5+5vRUss/wlb4t0gOGfh7Ej3E8HB6tHr481iV26xCxyWg
         g4pL7TYrI1pQJIblAYZ9vBfNUQPLzODNy/aEKYWmTbFTU+2jUfLklYAf23QLcaHwHeHG
         szr+izN1Cccz4m3M+bEhFAhBJtbFxzIiHBeQ8fKqAwp4aUbrIQ0kpPNAp3NbjSx15nF4
         PL8jC9G7LVeYdWykVprNJpBFpHh4FvoIGcqNsGbTBm0JRKmWTKeTCG05/woqOh3kvqXm
         CBJA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:from:to:subject:date:message-id;
        bh=CWCynePQ8um+5nRyCgMSCXhJ2HrwalLQWjg5pt3EE68=;
        b=RIBMnxEeCyOfb6DhFurDhEOchdsyG6CYhj0vSJKPb7m+len07lfBtGh4JMsYUsu8vz
         66qY8ERE8AIAH5NVzIAFJmzZerOZPfxKcjd+wP42D8Ngd5HVup+oo8tZaOHkqL9Zz1XC
         SykWbhXdUpsF7bNQer2hoebW2Qn2DB9tFfDxIZFf3IzAlwa3BOnPrvSltbRmkHh+j4Oy
         4SAwjGrW/ExtCbe2RIs8eBaMG3TNvO+EHbOj+d1ZENU+c7J5A2aDj3rXctY2F6pcBhQv
         hdDL1QzPr8sU/CnMWzO328gecu7lcJ9POp1zlwtNG9R+vdSxQYiFE8oidaVsS4RYuHMw
         r5rQ==
X-Gm-Message-State: APzg51D5N9su9sxrm+3yvoeHFWNQ9AU6ZXWDnvMIMQAkDnIS7DVKQrXA
        Gg7fQX/L62oh5kFdVYPxEVGXJm9n
X-Google-Smtp-Source: 
 ANB0Vdapa7UsqAdticdVq3jmNtk7UpZXnj7vXmkW6HciF4BC/Sy3PQL4n4OFkH6bO5smWuwymhBzPA==
X-Received: by 2002:adf:f504:: with SMTP id
 q4-v6mr20206694wro.241.1536677920630;
        Tue, 11 Sep 2018 07:58:40 -0700 (PDT)
Received: from donizetti.redhat.com
 (94-36-187-248.adsl-ull.clienti.tiscali.it. [94.36.187.248])
        by smtp.gmail.com with ESMTPSA id
 f17-v6sm23139382wrs.1.2018.09.11.07.58.39
        for <kvm@vger.kernel.org>
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Tue, 11 Sep 2018 07:58:39 -0700 (PDT)
From: Paolo Bonzini <pbonzini@redhat.com>
To: kvm@vger.kernel.org
Subject: [PATCH kvm-unit-tests] x86: umip: skip SMSW test because it fails
Date: Tue, 11 Sep 2018 16:58:38 +0200
Message-Id: <20180911145838.30503-1-pbonzini@redhat.com>
X-Mailer: git-send-email 2.17.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

SMSW does not cause a VMEXIT and therefore emulated UMIP fails.
Comment out the test.

Suggested-by: Liran Alon <liran.alon@oracle.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 x86/umip.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/x86/umip.c b/x86/umip.c
index cda1971..4566852 100644
--- a/x86/umip.c
+++ b/x86/umip.c
@@ -95,8 +95,11 @@ static void test_umip_gp(const char *msg)
 {
     puts(msg);
 
+#if 0
+    /* Skip this, because it cannot be emulated correctly.  */
     do_smsw();
     report("exception from smsw", gp_count == 1);
+#endif
     do_sgdt();
     report("exception from sgdt", gp_count == 1);
     do_sidt();
